arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationViewController.swift

```diff
@@ -0,0 +1,155 @@
+    private let maxScale = 2000.0
+    private let associationsOverlay = AGSGraphicsOverlay()
+    private let attachmentSymbol = AGSSimpleLineSymbol(style: .dot, color: .green, width: 5)
+    private let connectivitySymbol = AGSSimpleLineSymbol(style: .dot, color: .red, width: 5)
+    
+    func loadUtilityNetwork() {
+        utilityNetwork.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get all the edges and junctions in the network.
+                let sourcesByType = Dictionary(grouping: self.utilityNetwork.definition.networkSources) { $0.sourceType }
+                let operationalLayers = self.mapView.map?.operationalLayers
+                // Add all edges that are not subnet lines to the map.
+                let edgeLayers = sourcesByType[.edge]!
+                    .filter { $0.sourceUsageType != .subnetLine }
+                    .map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: edgeLayers)
+                // Add all the junctions to the map.
+                let junctionLayers = sourcesByType[.junction]!.map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: junctionLayers)
+                
+                // Create a renderer for the associations.
+                let attachmentValue = AGSUniqueValue(description: "Attachment", label: "", symbol: self.attachmentSymbol, values: [AGSUtilityAssociationType.attachment])
+                let connectivityValue = AGSUniqueValue(description: "Connectivity", label: "", symbol: self.connectivitySymbol, values: [AGSUtilityAssociationType.connectivity])
+                self.associationsOverlay.renderer = AGSUniqueValueRenderer(fieldNames: ["AssociationType"], uniqueValues: [attachmentValue, connectivityValue], defaultLabel: "", defaultSymbol: nil)
+                
+                self.createSwatches()
+                self.addAssociationGraphics()
+                self.mapViewDidChange()
+            }
+        }
+    }
+    
+    func addAssociationGraphics() {
+        // Check if the current viewpoint is outside of the max scale.
+        if let viewpoint = mapView.currentViewpoint(with: .centerAndScale),
+            viewpoint.targetScale >= maxScale {
+            return
+        }
+        
+        // Check if the current viewpoint has an extent.
+        if let viewpoint = mapView.currentViewpoint(with: .boundingGeometry) {
+            // Get all of the associations in extent of the viewpoint.
+            utilityNetwork.associations(withExtent: viewpoint.targetGeometry.extent) { [weak self] (associations, error) in
+                guard let self = self else { return }
+                if let error = error {
+                    print("Error loading associations: \(error)")
```

Comment:
It might be better to use `self.presentAlert(error:)` here, as user won't be able to see the error message in console.